src/data/Data.js

```diff
@@ -754,28 +759,25 @@ export const WorkspaceData = _.flow(
           .then((typesResult) => {
             setWdsTypes({ status: 'Ready', state: typesResult });
           })
-          .catch((err) => {
-            setWdsTypes({ status: 'Error', state: err });
+          .catch((error) => {
+            setWdsTypes({ status: 'Error', state: 'Error loading WDS schema' });
+            reportError('Error loading WDS schema', error);
           });
       },
       [signal]
     );
 
     const loadWdsData = useCallback(async () => {
-      try {
-        // Try to load the proxy URL
-        if (!wdsProxyUrl || wdsProxyUrl.status !== 'Ready') {
-          const wdsUrl = await loadWdsUrl(workspaceId);
-          if (wdsUrl) {
-            await loadWdsTypes(wdsUrl, workspaceId);
-          }
-        } else {
-          // If we have the proxy URL try to load the WDS types
-          const proxyUrl = wdsProxyUrl.state;
-          await loadWdsTypes(proxyUrl, workspaceId);
+      // Try to load the proxy URL
+      if (!wdsProxyUrl || wdsProxyUrl.status !== 'Ready') {
+        const wdsUrl = await loadWdsUrl(workspaceId);
+        if (wdsUrl) {
+          await loadWdsTypes(wdsUrl, workspaceId);
         }
-      } catch (error) {
-        console.log(`Error thrown loading WDS schema: ${error}`); // eslint-disable-line no-console
+      } else {
+        // If we have the proxy URL try to load the WDS types
+        const proxyUrl = wdsProxyUrl.state;
+        await loadWdsTypes(proxyUrl, workspaceId);
```

Comment:
Is there a way to rewrite this block so that we don't have to repeat the condition for `if (proxyUrl) { loadWdsTypes() }`?

Reply:
I felt the conditional logic here to be pretty hard to reason about, so I already started a followup PR to tighten it up a bit.  The strategy I'm trying is to decouple the status from the state, and use two different `useState` for each category of data loaded.  Hoping this helps comprehension but stay tuned.  I don't plan to address this as part of this PR though.